syntax = "proto3";

package location;

// The Location message
message Location {
    int32 id = 1;
    int32 person_id = 2;
    string longitude = 3;
    string latitude = 4;
    string creation_time = 5;
}

// Request for getting all locations
message GetAllLocationsRequest {}

// Response that contains a list of locations
message GetAllLocationsResponse {
    repeated Location locations = 1;
}

// The service that handles location operations
service LocationService {
    rpc CreateLocation (Location) returns (Location);

    // New RPC to get all locations
    rpc GetAllLocations (GetAllLocationsRequest) returns (GetAllLocationsResponse);
}